lib/dirac_mobius.cpp

```diff
@@ -186,17 +174,26 @@ namespace quda {
     // do nothing
   }
 
+  DiracMobiusPC::DiracMobiusPC(const DiracParam &param) : DiracMobius(param)
+  {
+    const int R[] = {2, 2, 2, 2};
+    extended_gauge = createExtendedGauge(*gauge, R, profile, true);
+  }
 
-  DiracMobiusPC::DiracMobiusPC(const DiracParam &param) : DiracMobius(param) {  }
-
-  DiracMobiusPC::DiracMobiusPC(const DiracMobiusPC &dirac) : DiracMobius(dirac) {  }
+  DiracMobiusPC::DiracMobiusPC(const DiracMobiusPC &dirac) : DiracMobius(dirac)
+  {
+    const int R[] = {2, 2, 2, 2};
+    extended_gauge = createExtendedGauge(*gauge, R, profile, true);
```

Comment:
Do we always need to create an extended gauge? Can we check for this on the first invocation of `MdagMlocal` and only create it then? We also need some way to check if the resident gauge has updated, though that can wait for a PR I need to work on for HISQ MG.

Reply:
@weinbe2 So I get around this with checking at the beginning of `MdagMLocal` and allocate `extended_gauge_field` if it doesn't exist yet. The problem with that is `MdagMLocal` is a `const` method and we are changing `extended_gauge_field` in its implementation. I have put `extended_gauge_field` `mutable` but let me know if you think that's too dirty a solution.